interpreter/golabels/integrationtests/golabels_integration_test.go

```diff
@@ -96,43 +98,45 @@ func Test_Golabels(t *testing.T) {
 				VerboseMode:            true,
 			})
 			require.NoError(t, err)
+			defer trc.Close()
 
 			trc.StartPIDEventProcessor(ctx)
-
-			err = trc.AttachTracer()
-			require.NoError(t, err)
+			require.NoError(t, trc.AttachTracer())
 
 			t.Log("Attached tracer program")
-
-			err = trc.EnableProfiling()
-			require.NoError(t, err)
-
-			err = trc.AttachSchedMonitor()
-			require.NoError(t, err)
+			require.NoError(t, trc.EnableProfiling())
+			require.NoError(t, trc.AttachSchedMonitor())
 
 			traceCh := make(chan *host.Trace)
+			require.NoError(t, trc.StartMapMonitors(ctx, traceCh))
 
-			err = trc.StartMapMonitors(ctx, traceCh)
-			require.NoError(t, err)
-
+			wg := sync.WaitGroup{}
+			wg.Add(1)
 			go func() {
```

Comment:
I'm wondering if we should rewrite this Go routine function with https://pkg.go.dev/golang.org/x/sync/errgroup to properly report the error. Calling panic works but might not be the best option to do so.

Reply:
So the panic is to get the backtraces of the goroutines the tracer creater. And they are not returning error, they are busy doing work or have hung. In either case panic gets the backtraces we want